test/spec/services/routesSpec.js

```diff
@@ -132,7 +133,25 @@ describe("RoutesService", function(){
       }];
       var warnings = RoutesService.getRouteWarnings(route, serviceTemplate);
       expect(warnings).toEqual(["Requested host www.example.com was rejected by the router. Reason: route bar already exposes www.example.com and is older."]);
-    });    
+    });
+
+    it("should warn if admitted route wildcardPolicy doesn't match", function() {
+      var route = angular.copy(routeTemplate);
+      route.spec.path = '/test';
+      route.spec.wildcardPolicy = 'Subdomain';
+      route.status.ingress = [{
+        host: 'www.example.com',
+        routerName: 'foo',
+        conditions: [{
+          type: "Admitted",
+          status: "True",
+          lastTransitionTime: "2016-02-17T17:18:51Z"
+        }],
+        wildcardPolicy: 'None'
+      }];
+      var warnings = RoutesService.getRouteWarnings(route, serviceTemplate);
+      expect(warnings).toEqual(['Router "foo" does not support wildcard subdomains. Your route will only be available at host www.example.com.']);
```

Comment:
The error message may be a bit different so you may want to check w/ older/newer router images.